### ISSUE TITLE
xml to otio flattern bug issue

### ISSUE THREAD

**Issue by cds1503** [PixarAnimationStudios/OpenTimelineIO#650]
Hello, 
I am reading the edited xml file with otio and flattening it. When I read a certain xml file, I confirmed that otio is properly created, but when I try flatten, the data disappears. Is there a way to debug this situation?

[Battle_Edit01g_S009-018_edl.xml.zip](https://github.com/PixarAnimationStudios/OpenTimelineIO/files/4247623/Battle_Edit01g_S009-018_edl.xml.zip)

**Comment by jminor** [comment]
One of the reasons we made `otioview` was to help debug issues like this. Are you able to run that tool to look visually at the original file and the flattened version?

**Comment by cds1503** [comment]
Yes, I could check the original data. After converting the original xml data to .otio, I checked it using otioview.
Then I went through the flattern algorithm and the data disappeared.
Eric Reinecke called it a bug in Slack.
So I registered the issue by registering it on github.

**Comment by jminor** [comment]
I see the same problem when I attempt to flatten your example file. The result has one video track and several audio tracks, but nothing in any of the clips. At first glance, I'm not sure what is wrong, but I would guess that something is clipping off the contents of the tracks because it incorrectly thinks they are out of the source_range.

One clue I see is that `otiostat` complains about one of the track lengths:
```parsed: True
top level object: Timeline.1
number of tracks: 24
There was a system error: track 2 is not the same duration as the other tracks. Track RationalTime(0, 1) duration, vs: RationalTime(22322, 23.976)
deepest nesting: 4
number of clips: 331
total duration: RationalTime(22322, 23.976)
There was a system error: invalid timecode rate
top level rate: 23.97602397602398
clips with cdl data: 0
Tracks with non standard types: 0
```

Are you able to put a breakpoint in the flatten code and walk through it? Or maybe you can try trimming down to a smaller example first.

**Comment by cds1503** [comment]
Debug results track 3,4,5 is an empty track.
So, except for 3,4,5, I flattened, and no blank timeline appeared.

I'm currently working with Python, so it's hard to debug the flatten algorithm.

**Comment by ssteinbach** [comment]
The underlying issue is that the flatten algorithm assumes that each track is the same duration.  When you have empty tracks, this breaks that assumption and the flatten algorithm falls down.  We should address this or provide better error reporting for the case that the algorithm doesn't handle.
